quinn-h3/benches/helpers.rs

```diff
@@ -44,7 +44,7 @@ impl Bench {
         let cert_chain = quinn::CertificateChain::from_certs(vec![cert.clone()]);
 
         let mut transport = quinn::TransportConfig::default();
-        transport.stream_window_bidi(102_400).unwrap();
+        transport.max_concurrent_bidi_streams(102_400).unwrap();
```

Comment:
Btw: I noticed that the old setting had kind of a memory problem: All the streams where pre-allocated, and if one used a high window (e.g. in order to let a different mechanism take care of limiting), the system run out of memory.

I guess this isn't resolved yet bis this change (which is totally fair). but also something that might be worthwhile to look into.

Reply:
What different mechanism could be used without exposing an endpoint to that same worst-case memory allocation under certain legal peer behavior?

Reply:
It's definitely the same worst-case allocation. But similar as with receive and transmit buffers the worst-case rarely happens, and services run under some kind of amortized expectations. 

If 95% of connections use only 10 streams, a nice amount of memory could be saved by not allocating the maximum upfront.

Reply:
The set of unopened but permitted remotely-initiated streams is sparse, so we'd need a `HashSet` or something for that. `size_of::<(Send, Recv)>()` = 248 is also a lot smaller than a typical I/O buffer. The savings could still be significant (16x?) for really huge stream budgets, I suppose, but I'm suspicious of the use case for such a budget.

Reply:
I see your point that it might matter more with many connections.